docs/tutorials/basics.md

```diff
@@ -0,0 +1,346 @@
+
+### CI build
+
+The current used CI flow to build the official docker container, you
+can see in [delivery.yaml](https://github.com/zalando/skipper/blob/master/delivery.yaml).
+Official release versions you will find at
+`registry.opensource.zalan.do/pathfinder/skipper:${RELEASE_VERSION}`,
+where `${RELEASE_VERSION}` is the git tag got by `$(git describe --tags --always --dirty)`.
+
+Test versions are released at
+`registry.opensource.zalan.do/pathfinder/skipper-test:${CDP_BUILD_VERSION}`
+for every pull request, limited to only repository members, because of
+compliance and security reasons.
+
+## Testing routes
+
+To test routes you can use a local build of skipper and pass arguments
+`-inline-routes=<route string>` or for more complex ones
+use a local `eskip` file on disk and use `-routes-file=<filepath>`.
+
+Example:
+
+```
+./bin/skipper -address :9999 -inline-routes 'r: * -> setQuery("lang", "pt") -> "http://127.0.0.1:8080/"'
+```
+
+Now you have a proxy running that will set a query to your request URL
+and call `http://127.0.0.1:8080/?lang=pt`
+
+The simplest way of testing a proxy is using a local backend and a
+local browser.
+
+Local backend example:
+
+```
+./bin/skipper -address :8080 -inline-routes 'r: * -> inlineContent("Hello world!") -> status(200) -> <shunt>'
+```
+
+If you want to do the request and see the response in detail, you can
+use `curl` as a browser, which should be installed on most Linux and
+Mac OS X computers.
+
+Example client call to our defined proxy:
+
+```
+% curl localhost:8080 -v
+* Rebuilt URL to: localhost:8080/
+*   Trying ::1...
+* Connected to localhost (::1) port 8080 (#0)
+> GET / HTTP/1.1
+> Host: localhost:8080
+> User-Agent: curl/7.49.0
+> Accept: */*
+>
+< HTTP/1.1 200 OK
+< Content-Length: 12
+< Content-Type: text/plain; charset=utf-8
+< Server: Skipper
+< Date: Thu, 01 Nov 2018 15:54:13 GMT
+<
+* Connection #0 to host localhost left intact
+Hello world!
+```
+
+## Current routing table
+
+To investigate the current routing table skipper has loaded into its
+memory, you can use the `-support-listener`, which defaults to port
+9911 and you have to do a GET request to the `/routes` endpoint.
+
+Example:
+
+```
+% curl localhost:9911/routes
+r: *
+  -> setQuery("lang", "pt")
+  -> "http://127.0.0.1:8000";
+```
+
+If you do not see your route, then you have most probably a syntax
+error in your route definition, such that the route was not loaded
+into memory.
+
+To print the number of routes, `X-Count` header, and the last update
+timestamp, `X-Timestamp` header, you can use a HEAD request to the
+support listener `/routes` endpoint:
+
+```bash
+% curl -I localhost:9911/routes
+HTTP/1.1 200 OK
+Content-Type: text/plain
+X-Count: 1
+X-Timestamp: 1541086036
+Date: Fri, 02 Nov 2018 00:30:43 GMT
+```
+
+For skipper operators the number of routes can be interesting for
+statistics and the timestamp to detect skipper instances that have not
+updated its routing table.
```

Comment:
I would add something like:

If there is more than 1024 routes used, then the paging the results is possible with the offset and limit query parameters:

```
curl locahost:9911/routes?offset=2048&limit=512
```